File: apps/api/AssetManagement.Api/Models/PersonHistoryChange.cs
namespace AssetManagement.Api.Models;

public class PersonHistoryChange
{
    public Guid Id { get; set; }

    public Guid PersonHistoryId { get; set; }
    public PersonHistory PersonHistory { get; set; } = null!;

    public string FieldName { get; set; } = string.Empty;
    public string? OldValue { get; set; }
    public string? NewValue { get; set; }
}

File: apps/api-kt/src/main/kotlin/com/assetmanagement/api/model/PersonHistoryChange.kt
package com.assetmanagement.api.model

import jakarta.persistence.*
import java.util.*

@Entity
@Table(name = "person_history_changes")
class PersonHistoryChange(
    @Id
    @Column(name = "id", columnDefinition = "CHAR(36)")
    var id: UUID = UUID.randomUUID(),

    @Column(name = "person_history_id", nullable = false, columnDefinition = "CHAR(36)")
    var personHistoryId: UUID = UUID.randomUUID(),

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "person_history_id", insertable = false, updatable = false)
    var personHistory: PersonHistory? = null,

    @Column(name = "field_name", nullable = false)
    var fieldName: String = "",

    @Column(name = "old_value", columnDefinition = "TEXT")
    var oldValue: String? = null,

    @Column(name = "new_value", columnDefinition = "TEXT")
    var newValue: String? = null
)
